### ISSUE TITLE
No reference type (parent of) Id is available to map as an include.

### ISSUE THREAD

**Issue by icnocop** [AutoMapper/AutoMapper.Extensions.OData#30]
Hi.

When I try to map a property of one type to another, an `InvalidOperationException` exception is thrown with the message `No reference type (parent of) Id is available to map as an include.`.

I've created a reproducible example here:
[TwoProjections.zip](https://github.com/AutoMapper/AutoMapper.Extensions.OData/files/4625148/TwoProjections.zip)

Steps to reproduce:

1. Open solution in Visual Studio 2017
2. Restore NuGet packages for solution
3. Rebuild solution (Debug | Any CPU)
4. Start both web applications in IIS Express (Set as startup project then Ctrl+F5)
5. Run all tests.
6. The tests in the `PersonBsTests` class should fail with the mentioned exception and message.

The queries are indicated in each test.
```
/PersonBs
/PersonBs?$top=10
/PersonBs?$top=10&$count=true
/PersonBs/$count
/PersonBs?$filter=contains(FirstName,'John')
```

```
System.InvalidOperationException: The mapped member PersonAId is of type System.Int64 and a child of the parameter type PrivateWebApplicationODataClient.PrivateWebApplication.Models.PersonA.  No reference type (parent of) PersonAId is available to map as an include.
   at AutoMapper.Extensions.ExpressionMapping.FindMemberExpressionsVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at AutoMapper.Extensions.ExpressionMapping.MapIncludesVisitor.<VisitMember>g__GetMappedMemberExpression|2_0(Expression parentExpression, List`1 propertyMapInfoList, <>c__DisplayClass2_0& )
   at AutoMapper.Extensions.ExpressionMapping.MapIncludesVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at AutoMapper.Extensions.ExpressionMapping.XpressionMapperVisitor.VisitUnary(UnaryExpression node)
   at System.Linq.Expressions.UnaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.<MapExpression>g__MapBody|5_1[TDestDelegate](Dictionary`2 typeMappings, XpressionMapperVisitor visitor, <>c__DisplayClass5_0`1& )
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.<MapExpression>g__CreateVisitor|5_0[TDestDelegate](Dictionary`2 typeMappings, <>c__DisplayClass5_0`1& )
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.MapExpression[TDestDelegate](IConfigurationProvider configurationProvider, LambdaExpression expression, Type typeSourceFunc, Type typeDestFunc, Func`3 getVisitor)
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.MapExpression[TDestDelegate](IMapper mapper, LambdaExpression expression, Func`3 getVisitor)
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.MapExpressionAsInclude[TDestDelegate](IMapper mapper, LambdaExpression expression)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at AutoMapper.Extensions.ExpressionMapping.MapperExtensions.MapIncludesList[TDestDelegate](IMapper mapper, IEnumerable`1 collection)
   at AutoMapper.AspNet.OData.QueryableExtensions.<GetAsync>d__4`2.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoMapper.AspNet.OData.QueryableExtensions.<GetAsync>d__1`2.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoMapper.AspNet.OData.QueryableExtensions.<>c__DisplayClass0_0`2.<<Get>b__0>d.MoveNext()
```

.NET Framework 4.6.1
AutoMapper 9.0.0
AutoMapper.AspNet.OData.EF6 1.0.3

This is a continuation of the issue reported in https://github.com/AutoMapper/AutoMapper.Extensions.ExpressionMapping/issues/71.

Thank you.

**Comment by BlaiseD** [comment]
Definitely a bug affecting EF6 piece.

**Comment by BlaiseD** [comment]
Fixed in [1.0.4-preview03](https://www.nuget.org/packages/AutoMapper.AspNetCore.OData.EFCore/1.0.4-preview03).

Note: /Entity/$count  is not supported (/Entity?$count=true is).  PRs welcome.
